src/modules/auth/auth.service.ts

```diff
@@ -285,14 +279,79 @@ export class AuthService {
     });
   }
 
+  async restorePassword(
+    authRestorePasswordOtpCode: AuthRestorePasswordDto,
+  ): Promise<void> {
+    const { email } = authRestorePasswordOtpCode;
+
+    const user = await this.usersService.findByEmail(email);
+
+    if (!user) {
+      throw new ConflictException('This email is not associated with any user');
+    }
+    const expiresIn = this.configService.get<string>(
+      'MAIL_RESET_PASSWORD_EXPIRATION_TIME',
+    );
+
+    const token = await this.token.generate<AuthPayloadToken>(
+      {
+        userId: user.id,
+      },
+      {
+        expiresIn,
+      },
+    );
+
+    const resetPasswordPageLink = `${this.configService.get<string>('CLIENT_RESET_PASSWORD-BASE_URL')}?${new URLSearchParams(
+      {
+        token,
+      },
+    ).toString()}`;
+
+    await this.mailerService.sendMail({
+      to: user.email,
+      subject: 'Black circle password restore',
+      template: 'restore-password',
+      context: {
+        name: user.name,
+        userEmail: user.email,
+        restorePasswordLink: resetPasswordPageLink,
+        expiresIn,
+      },
+    });
+  }
+
+  async resetPassword(authResetPasswordDto: AuthResetPasswordDto) {
```

Comment:
types

Reply:
I will add eslint rules for this, for avoid it in the future. Thx!